src/main/java/io/vertx/ext/mail/MailConfig.java

```diff
@@ -177,6 +169,19 @@ public MailConfig(JsonObject config) {
     userAgent = config.getString("userAgent", DEFAULT_USER_AGENT);
   }
 
+  @Override
+  public String getHostnameVerificationAlgorithm() {
+    // If not explicitly set then see if we should provide a default based on SMTP settings
+    String verification = super.getHostnameVerificationAlgorithm();
+    if ((verification == null || verification.isEmpty()) && !isTrustAll() &&
+        (isSsl() || starttls != StartTLSOptions.DISABLED)) {
+      // we can use HTTPS verification, which matches the requirements for SMTPS
+      verification = "HTTPS";
+    }
+    return verification;
```

Comment:
I am inclined to keep the `getter` method as simple as possible, and move the logic to the pool to call `setHostnameVerificationAlgorithm('HTTPS')`